### PR TITLE
Refactor Custom Transformation logic

### PR DESCRIPTION
**Pull request by sangm**
Adding tests for `option.using`

Previously, we were keeping state in two data structures `customTransformsOrder`
and `customTransforms`.

`customTransformsOrder` was used to maintain the order. We can normalize both of
these data structures by using a Map.

### CODE REVIEW

**Comment by sangm** [comment]
@kratiahuja Fixed tests :)

File: tests/unit/broccoli/ember-app-test.js [hunk 1]
```diff
@@ -884,6 +884,25 @@ describe('EmberApp', function() {
       expect(app._scriptOutputFiles['/assets/vendor.js']).to.contain('vendor/path/to/lib.js');
       expect(app._scriptOutputFiles['/assets/vendor.js']).to.contain('vendor/path/to/lib2.js');
     });
+
+    it('option.using throws exception given invalid inputs', function() {
+      // `using` is looped over if given, we should ensure this throws an exception with proper error message
+      expect(function() {
```
Comment (twokul) [inline tests/unit/broccoli/ember-app-test.js]: could we use arrow functions here?

**Comment by kratiahuja** [comment]
@rwjblue I don't have write access yet. Can you please merge? Thank you! :)

**Comment by kratiahuja** [comment]
Thanks @sangm !

**Comment by rwjblue** [comment]
🎉🎊👌

**Pull request merged**